Don't create a cursor when the bucket is empty

When we treat a DNS request and the bucket of records is still empty, we still create a Cursor. A Bbolt Cursor which iterate on an empty bucket will panic. So  we have to add a guard/filter to avoid this. 

NOTE: `easy to reproduce`

stacktrace
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x9d500a]

goroutine 12 [running]:
go.etcd.io/bbolt.(*Bucket).Cursor(...)
	/home/alessio/Projects/go/src/go.etcd.io/bbolt/bucket.go:84
main.getRecordsFromBucket(0x0, 0xc000306340, 0x3a, 0x7, 0x0, 0x0, 0x0, 0x0)
	/home/alessio/Projects/go/src/stream-dns/dns.go:52 +0x3a
main.findRecordsAndSetAsAnswersInMessage.func1(0xc0002d9500, 0x0, 0xc0002d9500)
	/home/alessio/Projects/go/src/stream-dns/dns.go:244 +0xeb
go.etcd.io/bbolt.(*DB).View(0xc00000cb40, 0xc00013d730, 0x0, 0x0)
	/home/alessio/Projects/go/src/go.etcd.io/bbolt/db.go:701 +0xa8
main.findRecordsAndSetAsAnswersInMessage(0xc000306340, 0x3a, 0x5, 0xc00000cb40, 0xc0000b6870, 0xc0000b67e0, 0xc000308a40, 0xd77fe0, 0xc000384360)
	/home/alessio/Projects/go/src/stream-dns/dns.go:240 +0xe6
main.registerHandlerForZone.func1(0xd841c0, 0xc00037c180, 0xc0000b67e0)
	/home/alessio/Projects/go/src/stream-dns/dns.go:172 +0x5ed
github.com/miekg/dns.HandlerFunc.ServeDNS(0xc0003c0080, 0xd841c0, 0xc00037c180, 0xc0000b67e0)
	/home/alessio/Projects/go/src/github.com/miekg/dns/server.go:52 +0x44
github.com/miekg/dns.(*ServeMux).ServeDNS(0xc000308300, 0xd841c0, 0xc00037c180, 0xc0000b67e0)
	/home/alessio/Projects/go/src/github.com/miekg/dns/serve_mux.go:128 +0x5d
github.com/miekg/dns.(*Server).serveDNS(0xc0003c6120, 0xc00037c180)
	/home/alessio/Projects/go/src/github.com/miekg/dns/server.go:665 +0x14e
github.com/miekg/dns.(*Server).serve(0xc0003c6120, 0xc00037c180)
	/home/alessio/Projects/go/src/github.com/miekg/dns/server.go:570 +0x31c
github.com/miekg/dns.(*Server).worker(0xc0003c6120, 0xc00037c180)
	/home/alessio/Projects/go/src/github.com/miekg/dns/server.go:244 +0x50
created by github.com/miekg/dns.(*Server).spawnWorker
	/home/alessio/Projects/go/src/github.com/miekg/dns/server.go:284 +0x83
```

--- a/dns.go
+++ b/dns.go
@@ -238,8 +238,11 @@ func recursionOnCname(b *bolt.Bucket, record Record) [][]Record {
 
 func findRecordsAndSetAsAnswersInMessage(qname string, qtype uint16, db *bolt.DB, m *dns.Msg, r *dns.Msg, metricsService *a.MetricsService) error {
 	err := db.View(func(tx *bolt.Tx) error {
-		// TODO: check if the bucket already exists and has keys
-		recordsBucket := tx.Bucket([]byte("records"))
+		recordsBucket, err := tx.CreateBucketIfNotExists([]byte("records"))
+
+		if err != nil {
+			return err
+		}
 
 		records, err := getRecordsFromBucket(recordsBucket, qname)
 

--- a/dns.go
+++ b/dns.go
@@ -238,11 +238,7 @@ func recursionOnCname(b *bolt.Bucket, record Record) [][]Record {
 
 func findRecordsAndSetAsAnswersInMessage(qname string, qtype uint16, db *bolt.DB, m *dns.Msg, r *dns.Msg, metricsService *a.MetricsService) error {
 	err := db.View(func(tx *bolt.Tx) error {
-		recordsBucket, err := tx.CreateBucketIfNotExists([]byte("records"))
-
-		if err != nil {
-			return err
-		}
+		recordsBucket := tx.Bucket([]byte("records"))
 
 		records, err := getRecordsFromBucket(recordsBucket, qname)
 

--- a/main.go
+++ b/main.go
@@ -127,6 +127,20 @@ func main() {
 		log.Fatal("database ", config.PathDB, err.Error(), "\nSet the environment variable: DNS_PATHDB")
 	}
 
+	err = db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucket([]byte("records"))
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		log.Panic(err.Error())
+	}
+
 	// Local Records
 	if config.LocalRecords != "" {
 		localRecords, err := localARecordsRawIntoRecords(config.LocalRecords)

